File: src/dataset/python/pig-latin_exercism-ThomasZumsteg/gemini/GeminiFlash_pig_latin_v1.py
import re

def translate(phrase):
    """translate converts a phrase to piglatin"""
    return ' '.join(translate_word(word) for word in phrase.split())

def translate_word(word):
    """translate_word converts a word to piglatin"""
    if not word:
        return word + "ay"

    if word.startswith('y') and not any(c in word[1:] for c in 'aeiou'):
        return word[1:] + word[0] + "ay"

    match = re.match(r'([^aeiou]*)([aeiou].*)', word)
    if match:
        consonants = match.group(1)
        vowels = match.group(2)
        if consonants:
            if consonants.endswith('q') and vowels.startswith('u'):
                return vowels + consonants + "ay"
            return vowels + consonants + "ay"
    return word + "ay"

File: src/dataset/Java/pig-latin_exercism-ThomasZumsteg/gemini/GeminiFlash_PigLatin_v1.java
public class PigLatin {

    private static final String VOWELS = "aeiou";

    public static String translate(String input) {
        String[] words = input.split(" ");
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < words.length; i++) {
            result.append(translateWord(words[i]));
            if (i < words.length - 1) {
                result.append(" ");
            }
        }

        return result.toString();
    }

    private static String translateWord(String input) {
        int firstVowelIndex = -1;
        for (int i = 0; i < input.length(); i++) {
            if (VOWELS.indexOf(input.charAt(i)) != -1) {
                firstVowelIndex = i;
                break;
            }
            if (input.charAt(i) == 'y') {
                if (i == 0) continue;
                firstVowelIndex = i;
                break;
            }
        }

        if (firstVowelIndex == -1) {
            return input + "ay";
        }

        if (input.startsWith("qu")) {
            return input.substring(2) + "quay";
        }

        for (int i = 0; i < input.length() - 1; i++) {
            if (input.substring(i, i + 2).equals("qu") && (i < firstVowelIndex)) {
                return input.substring(i + 2) + input.substring(0, i + 2) + "ay";
            }
        }

        if (firstVowelIndex == 0) {
            return input + "ay";
        }

        return input.substring(firstVowelIndex) + input.substring(0, firstVowelIndex) + "ay";
    }
}